crates/storage/store_db/libmdbx.rs

```diff
@@ -1273,29 +1272,53 @@ table!(
 
 table!(
     /// Stores blocks that are pending validation.
-    ( PendingBlocks ) BlockHashRLP => BlockRLP
+    ( PendingBlocks ) BlockHashKey => BlockRLP
 );
 
 table!(
     /// State Snapshot used by an ongoing sync process
-    ( StateSnapShot ) AccountHashRLP => AccountStateRLP
+    ( StateSnapShot ) AccountHashKey => AccountStateRLP
 );
 
 dupsort!(
     /// Storage Snapshot used by an ongoing sync process
-    ( StorageSnapShot ) AccountHashRLP => (AccountStorageKeyBytes, AccountStorageValueBytes)[AccountStorageKeyBytes]
+    ( StorageSnapShot ) AccountHashKey => (AccountStorageKeyBytes, AccountStorageValueBytes)[AccountStorageKeyBytes]
 );
 
 table!(
     /// Storage trie paths in need of healing stored by hashed address
-    ( StorageHealPaths ) AccountHashRLP => TriePathsRLP
+    ( StorageHealPaths ) AccountHashKey => TriePathsRLP
 );
 
 table!(
     /// Stores invalid ancestors
-    ( InvalidAncestors ) BlockHashRLP => BlockHashRLP
+    ( InvalidAncestors ) BlockHashKey => BlockHashRLP
 );
 
+/// Wrapper for a H256 used as a key
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct WrappedH256(H256);
+
+impl Encodable for WrappedH256 {
+    type Encoded = [u8; 32];
+
+    fn encode(self) -> Self::Encoded {
+        self.0.0
+    }
+}
+
+impl Decodable for WrappedH256 {
+    fn decode(b: &[u8]) -> anyhow::Result<Self> {
+        Ok(WrappedH256(H256(b.try_into()?)))
+    }
+}
+
+impl From<H256> for WrappedH256 {
+    fn from(value: H256) -> Self {
+        Self(value)
+    }
+}
```

Comment:
You can do `impl Into<H256> for WrappedH256` to get the reciprocal operation while complying with the orphan rule. So you can use `into()` in both cases.